shell/run_shell_cmd.go

```diff
@@ -46,51 +44,41 @@ func RunTerraformCommandAndCaptureOutput(terragruntOptions *options.TerragruntOp
 func RunShellCommandWithOutput(terragruntOptions *options.TerragruntOptions, command string, args ...string) (string, error) {
 	terragruntOptions.Logger.Printf("Running command: %s %s", command, strings.Join(args, " "))
 
-	var outToErr bool = false
+	var outBuf bytes.Buffer
 
 	cmd := exec.Command(command, args...)
 
-	stdoutIn, err := cmd.StdoutPipe()
-	if err != nil {
-		return "", errors.WithStackTrace(err)
-	}
-
-	stderrIn, err := cmd.StderrPipe()
-	if err != nil {
-		return "", errors.WithStackTrace(err)
-	}
-
 	// TODO: consider adding prefix from terragruntOptions logger to stdout and stderr
 	cmd.Stdin = os.Stdin
 	cmd.Env = toEnvVarsList(terragruntOptions.Env)
 
+	var errWriter = terragruntOptions.ErrWriter
+	var outWriter = terragruntOptions.Writer
 	// Terragrunt can run some commands (such as terraform remote config) before running the actual terraform
 	// command requested by the user. The output of these other commands should not end up on stdout as this
 	// breaks scripts relying on terraform's output.
 	if !reflect.DeepEqual(terragruntOptions.TerraformCliArgs, args) {
-		outToErr = true
+		outWriter = terragruntOptions.ErrWriter
 	}
 
 	cmd.Dir = terragruntOptions.WorkingDir
+	// Inspired by https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html
+	cmd.Stderr = io.MultiWriter(errWriter, &outBuf)
+	cmd.Stdout = io.MultiWriter(outWriter, &outBuf)
 
 	if err := cmd.Start(); err != nil {
 		// bad path, binary not executable, &c
 		return "", errors.WithStackTrace(err)
 	}
 
-	output, err := readStdoutAndStderr(stdoutIn, stderrIn, terragruntOptions, outToErr)
-	if err != nil {
-		return "", errors.WithStackTrace(err)
-	}
-
 	cmdChannel := make(chan error)
 	signalChannel := NewSignalsForwarder(forwardSignals, cmd, terragruntOptions.Logger, cmdChannel)
 	defer signalChannel.Close()
 
-	err = cmd.Wait()
+	err := cmd.Wait()
 	cmdChannel <- err
 
-	return output, errors.WithStackTrace(err)
+	return string(outBuf.Bytes()), errors.WithStackTrace(err)
```

Comment:
Sorry, one more tiny thing: I think you can do `outBuf.String()`

Reply:
Fixed.